### ISSUE TITLE
Workflow: filter based on file path

### ISSUE THREAD

**Issue by martin-rueegg** [nextcloud/server#22976]
#### Context
When creating (work)flows, there are different filters that can be used in a rule, e.g. [file name](https://github.com/nextcloud/server/blob/master/apps/workflowengine/lib/Check/FileName.php) or [file size](https://github.com/nextcloud/server/blob/master/apps/workflowengine/lib/Check/FileSize.php), etc. However, there is no filtering possible on _file path._

#### Use case
For example, on my mobile phone I upload pictures with AutoUpload to a specific folder on the server. Now I want to downsize all those pictures into another folder and saving the original in a different folder again, leaving the upload folder empty. I have a script for that. All I need to do is running that script whenever a new file is uploaded or moved into that folder.

#### Alternative?
While I can use a RegEx for example to match on the file extension (using the [`File name`-filter](https://github.com/nextcloud/server/blob/bdc60ef9b2d4793b3da7887632fb4b0e77bf964a/apps/workflowengine/src/components/Checks/file.js#L38)), I cannot do the same for the path, as that information is not available (been cut off by [`basename()`](https://github.com/nextcloud/server/blob/bdc60ef9b2d4793b3da7887632fb4b0e77bf964a/apps/workflowengine/lib/Check/FileName.php#L50)).

### Solution

I have created a solution that works on NC17 (but needs review and might need some tweaks) and am now struggling to get it working on NC18+ (testing with 19.0.2). The main issue is, that while I have the filter in place, I don't manage to get it in the frontend into the list of available filters.

#### Help needed
Help on finalizing the patch would be great so that I can open a PR (guidance on the latter will also be appreciated once everything is ready).

PS: happy to provide a filter for https://github.com/nextcloud/server/issues/5330 as well, although I don't really see the use case on that.... :-)

------

#### NC17

- [branch](https://github.com/metaworx/nextcloud-server/tree/add/workflowengine/file-path/stable17)
- [patch](https://github.com/nextcloud/server/compare/stable17...metaworx:add/workflowengine/file-path/stable17?expand=1)

I haven't included the compiled version of [workflowengine.js](https://github.com/metaworx/nextcloud-server/blob/add/workflowengine/file-path/stable17/apps/workflowengine/src/workflowengine.js) in order to keep merging simple. Please run `make` first.

#### master

- [branch](https://github.com/metaworx/nextcloud-server/tree/add/workflowengine/file-path/master)
- [patch](https://github.com/nextcloud/server/compare/master...metaworx:add/workflowengine/file-path/master?expand=1)

**Comment by martin-rueegg** [comment]
Ping @juliushaertl, @blizzz

**Comment by martin-rueegg** [comment]
- https://github.com/nextcloud/server/pull/23451
- https://github.com/nextcloud/server/pull/23452

**Comment by kesselb** [comment]
Duplicate of https://github.com/nextcloud/server/issues/19810
